event-sourcing-introduction.md

```diff
@@ -0,0 +1,232 @@
+# Event-Sourcing: A Different Way to Model State
+
+Event-Sourcing is an architectural pattern where we store the history of state changes as a sequence of events, rather than just storing the current state. Let's explore this concept using a relatable example: growing marijuana plants.
+
+## The Scenario
+
+I have a marijuana plant. My grandma has another marijuana plant. Mine is starving and struggling to grow, while the one my grandma tends to is flourishing. Why? Let's see how Event-Sourcing can help us understand the difference.
+
+## CRUD: The Traditional Approach
+
+In a traditional CRUD (Create, Read, Update, Delete) system, we only store the current state:
+
+```typescript
+interface Plant {
+  id: string;
+  ownerId: string;
+  height: number;
+  health: number;
+  lastWatered: Date;
+  trimCount: number;
+  isHarvested: boolean;
+}
+
+class PlantRepository {
+  async update(plant: Plant): Promise<void> {
+    // Just overwrite the current state
+    await database.save(plant);
+  }
+}
+```
+
+With CRUD, if we look at our plants right now:
+
+```typescript
+const myPlant: Plant = {
+  id: "plant-1",
+  ownerId: "me",
+  height: 15,
+  health: 30, // Poor health!
+  lastWatered: new Date("2023-10-15"),
+  trimCount: 0,
+  isHarvested: false
+};
+
+const grandmasPlant: Plant = {
+  id: "plant-2",
+  ownerId: "grandma",
+  height: 45,
+  health: 95, // Excellent health!
+  lastWatered: new Date("2023-10-30"),
+  trimCount: 3,
+  isHarvested: false
+};
+```
+
+**The Problem**: We can see my plant is unhealthy, but we don't know *why*. We've lost all the history. Was it watered regularly? Was it ever trimmed? We can't tell because we only have the current snapshot.
```

Comment:
My grandma did better. We can see that clearly.
But WHY?

make a better transistion to event sourcing.